src/library/filescanner.c

```diff
@@ -1600,6 +1614,302 @@ scan_metadata(const char *path, struct media_file_info *mfi)
   return LIBRARY_PATH_INVALID;
 }
 
+static const char *
+virtual_path_to_path(const char *virtual_path)
+{
+  if (strncmp(virtual_path, "/file:", strlen("/file:")) == 0)
+    return virtual_path + strlen("/file:");
+
+  if (strncmp(virtual_path, "file:", strlen("file:")) == 0)
+    return virtual_path + strlen("file:");
+
+  return NULL;
+}
+
+static bool
+check_path_in_directories(const char *path)
```

Comment:
I added a check, so that paths containing "/../" are invalid. realpath only works if the file already exists. And i guess symlinks are ok for directories in the library directory.

I hope this is good enough. But i am open to further suggestions. It should not be too critical as we always append ".m3u" to the given path. So only m3u files can be modified/created/deleted.